pkg/mcp/snapshot/inmemory.go

```diff
@@ -24,11 +24,14 @@ import (
 	"github.com/gogo/protobuf/proto"
 	"github.com/gogo/protobuf/types"
 
+	"sync"
+
 	mcp "istio.io/api/mcp/v1alpha1"
 )
 
 // InMemory Snapshot implementation
 type InMemory struct {
+	mutex     sync.RWMutex
```

Comment:
There shouldn't be a need for mutexes in the snapshot code. It uses builder/immutable data pattern.

Reply:
ok